internal/zinc-core/src/main/scala/sbt/internal/inc/Lookup.scala

```diff
@@ -7,40 +7,23 @@ import xsbti.compile.{ CompileAnalysis, ExternalHooks }
 /**
  * A trait that encapsulates looking up elements on a classpath and looking up
  * an external (for another subproject) Analysis instance.
- *
- * Multiple variants of `lookupAnalysis` exists because different parts of zinc
- * lookup an external Analysis a bit differently. In the future, only the
- * variant that takes binary class name as a parameter should be kept. This
- * variant can be implemented without an expensive classpath lookup.
- * See https://github.com/sbt/sbt/issues/2525
  */
 trait Lookup extends ExternalLookup {
   /**
-   * Lookup an element on the classpath corresponding to a given binary class name.
-   * If found class file is stored in a jar file, the jar file is returned.
-   *
-   * @param binaryClassName
-   * @return
+   * Returns the current classpath if the classpath has changed from the last compilation.
    */
-  def lookupOnClasspath(binaryClassName: String): Option[File]
+  def changedClasspath: Option[Vector[File]]
```

Comment:
Yea. Those are two different concerns. Some() vs None is yes it changed, or no it hasn't.
Vector contains the actual value. I don't like encoding extra meaning to magic values like empty.